examples/cluster/plot_hdbscan.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+"""
+====================================
+Demo of HDBSCAN clustering algorithm
+====================================
+.. currentmodule:: sklearn
+
+In this demo we will take a look at :class:`cluster.HDBSCAN` from the
+perspective of generalizing the :class:`cluster.DBSCAN` algorithm.
+We'll compare both algorithms on specific datasets. Finally we'll evaluate
+HDBSCAN's sensitivity to certain hyperparameters. We first define a couple
+utility functions for convenience.
+"""
+# %%
+import numpy as np
+
+from sklearn.cluster import HDBSCAN, DBSCAN
+from sklearn.datasets import make_blobs
+import matplotlib.pyplot as plt
+
+
+def plot(X, labels, probabilities=None, parameters=None, ground_truth=False, ax=None):
+    if ax is None:
+        _, ax = plt.subplots(figsize=(10, 4))
```

Comment:
I think having the same scale on graph to visually represent isotropy of the points' distribution's modes is appropriate. This is I think a good proposal to use the same scale, but there might be a better one (excuse me for being a better cook than a matplotlib user).  
```suggestion
        _, ax = plt.subplots(figsize=(10, 10))
```

Reply:
This would require a significant amount of space on the page for each plot, and wouldn't affect the grouped plots.

IMO since the isotropy of the distributions is not relevant (and not required at all for (H)DBSCAN) we should leave it as-is for now. If there are strong opinions for altering it, I'll happily do so, but would prefer to keep it as-is for now.